### ISSUE TITLE
fn:serialize returns emtp for arrays

### ISSUE THREAD

**Issue by line-o** [eXist-db/exist#2339]
### What is the problem

```xquery
fn:serialize([1])
```

returns ''

### What did you expect
> Describe what you expected to happen. Add for example a reference to a [specification](https://www.w3.org/TR/xquery-3/).

### Describe how to reproduce or add a test
> Describe how we can can reproduce the problem.

> The *best* way is to provide a 'self containing' test, a test that can be run without dependancies. The [XQSuite - Annotation-based Test Framework for XQuery](http://exist-db.org/exist/apps/doc/xqsuite.xml) makes it very easy for you to create tests. These tests can be executed from the [eXide editor](http://exist-db.org/exist/apps/eXide/index.html) (XQuery - Run as Test)

### Context information
Please always add the following information
- eXist-db version + Git Revision hash e.g. eXist-db 3.0 /  acd0c14 
- Java version (e.g. Java8u121)
- Operating system (Windows 7, Linux, MacOs)
- 32 or 64 bit
- How is eXist-db installed? (JAR installer, DMG, WAR file, clone from GitHub)
- Any custom changes in e.g. conf.xml

**Comment by line-o** [comment]
I am also wondering what the expected behaviour for maps is.
BaseX just throws an error while existdb will return an empty string again. 
When I have time to look at the spec I might expand this issue.

**Comment by line-o** [comment]
Same behaviour applies to `util:log([1,2])`

**Comment by dizzzz** [comment]
maybe in `fn:serialize([1])` the [1] is seen as a predicate.....

saxon returns "1" anyway

**Comment by joewiz** [comment]
`[1] instance of array(*), [1,2] instance of array(*)` returns `true(), true()`, so eXist knows it's an array. 

The spec for `fn:serialize` says:

> If the second argument is omitted, or is supplied in the form of an output:serialization-parameters element, then the values of any serialization parameters that are not explicitly specified is ·implementation-defined·, and may depend on the context.

This leads to the question, what is eXist's default set of serialization parameters?

If, however, we supply a serialization method, then eXist should respect it. Let's try "text":

```code
xquery version "3.1";

serialize([1], map {"method": "text"})
```

This also returns "". Thus I think we have a serialization bug. The relevant portion of the spec is https://www.w3.org/TR/xslt-xquery-serialization-31/#serdm, which states:

> Each item in the sequence that is an array is flattened by calling the function `array:flatten()` before being copied.

If we do this manually (via `serialize(array:flatten([1]))`, we get the expected result, "1".

In other words, eXist is failing to flatten arrays during sequence normalization.

**Comment by line-o** [comment]
Thanks @joewiz for your thorough investigation of the issue. I already learned a lot.

**Comment by line-o** [comment]
When serializing maps 

```xquery
xquery version "3.1";
let $m := map{"key":"1"}
return (
  serialize($m),
  serialize($m, map{'method':'text'}),
  serialize($m, map{'method':'json'}),
  serialize($m, map{'method':'xml'})
)

(: yields ("", "", "{""key"": ""1""}", "") :)
```
They all seem to be broken, if the JSON serialisation really does add quotes to key and value.

**Comment by line-o** [comment]
Digging a little deeper I found that there is an 'adaptive' method which does things as one would expect
```xquery
xquery version "3.1";
let $m := (map {"key":"1"}, [1, [2, 3]])
return (
  serialize($m),
  serialize($m, map {'method':'adaptive'}),
  serialize($m, map {'method':'text'}),
  serialize($m, map {'method':'json'}),
  serialize($m, map {'method':'xml'})
)
```

**Comment by line-o** [comment]
Just read the spec for `array:flatten` to learn that is indeed a function that returns a sequence of all members of nested arrays.

**Comment by joewiz** [comment]
This problem still affects eXist 5.2.0.

**Comment by line-o** [comment]
I revisited my feature request and the corresponding [XQuery FO 3.1 spec for fn:serialize](https://www.w3.org/TR/xpath-functions-31/#func-serialize).

The spec clearly states that the default for the "method" option is "xml". Closing this issue as wont fix
